.github/workflows/build_and_functional_tests.yml

```diff
@@ -0,0 +1,52 @@
+name: Build and run functional tests using ragger through reusable workflow
+
+# This workflow will build the app and then run functional tests using the Ragger framework upon Speculos emulation.
+# It calls a reusable workflow developed by Ledger's internal developer team to build the application and upload the
+# resulting binaries.
+# It then calls another reusable workflow to run the Ragger tests on the compiled application binary.
+#
+# While this workflow is optional, having functional testing on your application is mandatory and this workflow and
+# tooling environment is meant to be easy to use and adapt after forking your application
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+      - main
+      - develop
+  pull_request:
+
+jobs:
+  build_plugin:
+    name: Build plugin using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      app_repository: LedgerHQ/app-plugin-boilerplate
+      flags: "DEBUG=1"
+      upload_app_binaries_artifact: libraries_binaries
+      upload_as_lib_artifact: PluginBoilerplate
+      run_for_devices: '["nanos", "nanox", "nanosp", "stax"]'
+
+  build_ethereum_app:
+    name: Build Ethereum app using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      app_repository: LedgerHQ/app-ethereum
+      app_branch_name: develop
```

Comment:
develop or master ? I think it's important to test both, we might want to add the two

Reply:
yes it is a good question !
I guess it will depend on the plugin branch (develop or master)

Reply:
I think it does not depend, when we develop a new version of the plugin (develop branch), we want it to be compatible with the currently deployed ethereum (master branch) and the next deployed ethereum (develop branch)
We should add both, I think

Reply:
The CI will never really run on the master branch of the plugin